bson/typings.py

```diff
@@ -24,6 +24,6 @@
 
 # Common Shared Types.
 _DocumentOut = Union[MutableMapping[str, Any], "RawBSONDocument"]
-_DocumentType = TypeVar("_DocumentType", bound=Mapping[str, Any])
+_DocumentType = TypeVar("_DocumentType", bound=Mapping[str, Any], covariant=True)
```

Comment:
Hmm why does this need covariant=True now?

Reply:
Related: https://github.com/mongodb/mongo-python-driver/pull/1318/files#r1284910163